### PR TITLE
NO-JIRA: Move secrets-store-csi-driver-operator to rhel9

### PR DESCRIPTION
**Pull request by dobsonj**
Needed for https://github.com/openshift-eng/ocp-build-data/pull/4396
/cc @openshift/storage

### CODE REVIEW

**Comment by dobsonj** [comment]
Actually I think there is no need for the hold, the ART bot should not try to update these files while the build is disabled (I think)
/unhold

File: Dockerfile.mustgather [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM registry.ci.openshift.org/ocp/4.16:cli
+FROM registry.ci.openshift.org/ocp/4.16:base-rhel9
 RUN dnf install -y --nodocs --setopt=install_weak_deps=False openshift-clients \
```
Comment (gnufied) [inline Dockerfile.mustgather:2]: Is that intentional? Changing from `cli` base to `base-rhel9` ?

Comment (dobsonj) [reply Dockerfile.mustgather:2]: Yeah it was intentional, Joep previously moved it from the cli image to base-rhel9 and added the dnf install line for openshift-clients [here](https://github.com/openshift/secrets-store-csi-driver-operator/commit/3192537b3d1474621d2415e2b4adbe1509e49b7c). I saw other places where this was done as well.

**Pull request merged**